cmake/match.cmake

```diff
@@ -6,32 +6,64 @@
 # with a known good match file.
 #
 
-if(NOT DEFINED TEST_FILE)
-    message(FATAL_ERROR "TEST_FILE needs to be defined")
-endif()
-if(NOT DEFINED MATCH_FILE)
-    message(FATAL_ERROR "MATCH_FILE needs to be defined")
+# Process variables passed to the script
```

Comment:
I'm not sure I like it this way. I'd rather have just one way of building tests with match files. But let's leave it for now.